fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java

```diff
@@ -189,11 +200,33 @@ public Response head() {
 
             // we set the content-type explicitly to avoid content-negotiation from getting in the way
             builder.type(mediaType.toString());
+        } else {
+            final MediaType mediaType = RDFMediaType.TURTLE_TYPE;
+            final RdfStream stream = getResourceAsStream();
```

Comment:
@dbernstein personally, I think we should close the issue as "Won't Fix". IMO, the only viable alternative is to _not_ include a Content-Length header for HEAD requests on containers. When I looked into this ticket earlier, I did not see a way to do that, as Jersey will (helpfully) add a Content-Length header for you.

Even if the performance considerations have been addressed, a HEAD request should, under no circumstances, try to fetch and/or serialize an entire object -- it should only check whether the object exists along with some of the basic LDP types.